### ISSUE TITLE
Could not find recipe default. Anyway I'm trying

### ISSUE THREAD

**Issue by makarek** [bloomberg/chef-umami#26]
Hello

I wanted to give it a try, and I'm always getting same error about missing recipe.
Symptoms below:

```
[root@cocojumbo umami_test]# tree
.
├── metadata.rb
├── Policyfile.rb
└── recipes
    └── default.rb

1 directory, 3 files

[root@cocojumbo umami_test]# chef install
Building policy test-policy
Expanded run list: recipe[umami_test]
Caching Cookbooks...
Installing umami_test >= 0.0.0 from path

Lockfile written to /root/chef-repo/cookbooks/umami_test/Policyfile.lock.json
Policy revision id: d5791fe56fcf04fbfc09ee0369893b7dbaf8caacf4b03ed2b401b98585eed2ee

[root@cocojumbo umami_test]# chef exec umami

Exporting the policy, related cookbooks, and a valid client configuration...
Generating a new Policyfile from 'Policyfile.rb'...
Installing cookbooks from lock
Installing umami_test 0.1.11

Uploading the policy and related cookbooks...
Uploading policy to policy group local
Uploaded umami_test 0.1.11 (ae24e2fb)

Executing chef-client compile phase...
Using policy 'test-policy' at revision '7adf055288eb9863b10a85d5371af7f4f2a1f43c99918c2d28276fef7081285d'
resolving cookbooks for run list: ["umami_test::default@0.1.11 (ae24e2f)"]
Synchronizing Cookbooks:
Installing Cookbook Gems:
Compiling Cookbooks...

================================================================================
Recipe Compile Error
================================================================================

Chef::Exceptions::RecipeNotFound
--------------------------------
could not find recipe default for cookbook umami_test

System Info:
------------
chef_version=13.8.5
platform=redhat
platform_version=7.4
ruby=ruby 2.4.3p205 (2017-12-14 revision 61247) [x86_64-linux]
program_name=/root/.chefdk/gem/ruby/2.4.0/bin/umami
executable=/root/.chefdk/gem/ruby/2.4.0/bin/umami

/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.8.5/lib/chef/cookbook_version.rb:194:in `load_recipe': could not find recipe default for cookbook umami_test (Chef::Exceptions::RecipeNotFound)
        from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.8.5/lib/chef/run_context.rb:342:in `load_recipe'
        from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.8.5/lib/chef/run_context/cookbook_compiler.rb:163:in `block in compile_recipes'
        from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.8.5/lib/chef/run_context/cookbook_compiler.rb:160:in `each'
        from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.8.5/lib/chef/run_context/cookbook_compiler.rb:160:in `compile_recipes'
        from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.8.5/lib/chef/run_context/cookbook_compiler.rb:77:in `compile'
        from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.8.5/lib/chef/run_context.rb:191:in `load'
        from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.8.5/lib/chef/policy_builder/policyfile.rb:188:in `setup_run_context'
        from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.8.5/lib/chef/client.rb:513:in `setup_run_context'
        from /root/.chefdk/gem/ruby/2.4.0/gems/chef-umami-0.0.6/lib/chef-umami/client.rb:38:in `compile'
        from /root/.chefdk/gem/ruby/2.4.0/gems/chef-umami-0.0.6/lib/chef-umami/runner.rb:95:in `run'
        from /root/.chefdk/gem/ruby/2.4.0/gems/chef-umami-0.0.6/bin/umami:7:in `<top (required)>'
        from /root/.chefdk/gem/ruby/2.4.0/bin/umami:23:in `load'
        from /root/.chefdk/gem/ruby/2.4.0/bin/umami:23:in `<main>'
```

Am I doing something wrong ?

**Comment by RyanFrantz** [comment]
Does `recipes/default.rb` still exist? I created a similarly sparse cookbook directory and ran `chef exec umami` with no issues. When I deleted `recipes/default.rb` and ran `chef exec umami` again, I received the same error as what you experienced.

**Comment by makarek** [comment]
Hi 

Yes, it exists and works:

```
[root@cocojumbo recipes]# chef-client -z -o umami_test
[2018-06-04T15:02:21+02:00] INFO: Started chef-zero at chefzero://localhost:1 with repository at /root/chef-repo
  One version per cookbook

Starting Chef Client, version 14.1.12
[2018-06-04T15:02:21+02:00] INFO: *** Chef 14.1.12 ***
[2018-06-04T15:02:21+02:00] INFO: Platform: x86_64-linux
[2018-06-04T15:02:21+02:00] INFO: Chef-client pid: 4980
[2018-06-04T15:02:21+02:00] INFO: The plugin path /etc/chef/ohai/plugins does not exist. Skipping...
[2018-06-04T15:02:24+02:00] WARN: Run List override has been provided.
[2018-06-04T15:02:24+02:00] WARN: Original Run List: []
[2018-06-04T15:02:24+02:00] WARN: Overridden Run List: [recipe[umami_test]]
[2018-06-04T15:02:24+02:00] INFO: Run List is [recipe[umami_test]]
[2018-06-04T15:02:24+02:00] INFO: Run List expands to [umami_test]
[2018-06-04T15:02:24+02:00] INFO: Starting Chef Run for 174287
[2018-06-04T15:02:24+02:00] INFO: Running start handlers
[2018-06-04T15:02:24+02:00] INFO: Start handlers complete.
resolving cookbooks for run list: ["umami_test"]
[2018-06-04T15:02:25+02:00] INFO: Loading cookbooks [umami_test@0.1.11]
[2018-06-04T15:02:25+02:00] INFO: Skipping removal of obsoleted cookbooks from the cache
Synchronizing Cookbooks:
[2018-06-04T15:02:25+02:00] INFO: Storing updated cookbooks/umami_test/recipes/default.rb in the cache.
  - umami_test (0.1.11)
Installing Cookbook Gems:
Compiling Cookbooks...
Converging 1 resources
Recipe: umami_test::default
  * yum_package[openssh-server] action install[2018-06-04T15:02:25+02:00] INFO: Processing yum_package[openssh-server] action install (umami_test::default line 6)
 (up to date)
[2018-06-04T15:02:26+02:00] WARN: Skipping final node save because override_runlist was given
[2018-06-04T15:02:26+02:00] INFO: Chef Run complete in 2.372677278 seconds
[2018-06-04T15:02:26+02:00] INFO: Skipping removal of unused files from the cache

Running handlers:
[2018-06-04T15:02:26+02:00] INFO: Running report handlers
Running handlers complete
[2018-06-04T15:02:26+02:00] INFO: Report handlers complete
Chef Client finished, 0/1 resources updated in 05 seconds

```

But still getting same error after chef exec umami

**Comment by makarek** [comment]
I laso spoke with other collegues in my org to check and they also run into this issue

**Comment by Santosh-Y-Jadhav** [comment]
I am also facing same issue 
[santosh_jadhav@devrls820srv3 santosh]$ sudo chef-client -z -c .chef/config.rb
Starting Chef Client, version 14.10.9
Using policy 'santosh' at revision 'f26d6885c16ffe0b7d031d37eeaf55618da6023d83c3620f22e4deafbf55d790'
resolving cookbooks for run list: ["santosh::default@1.0.0 (c985127)"]
Synchronizing Cookbooks:
  - santosh (1.0.0)
Installing Cookbook Gems:
Compiling Cookbooks...
Converging 2 resources
Recipe: santosh::default
  * linux_user[wutang] action create (up to date)
  * directory[/usr/local/wutang] action create (up to date)

Running handlers:
Running handlers complete
Chef Client finished, 0/2 resources updated in 01 seconds
[santosh_jadhav@devrls820srv3 santosh]$


Compiling Cookbooks...

================================================================================
Recipe Compile Error
================================================================================

Chef::Exceptions::RecipeNotFound
--------------------------------
could not find recipe default for cookbook santosh

System Info:
------------
chef_version=13.12.14
platform=centos
platform_version=6.10
ruby=ruby 2.5.3p105 (2018-10-18 revision 65156) [x86_64-linux]
program_name=/root/.chefdk/gem/ruby/2.5.0/bin/umami
executable=/root/.chefdk/gem/ruby/2.5.0/bin/umami

Traceback (most recent call last):
	13: from /root/.chefdk/gem/ruby/2.5.0/bin/umami:23:in `<main>'
	12: from /root/.chefdk/gem/ruby/2.5.0/bin/umami:23:in `load'
	11: from /root/.chefdk/gem/ruby/2.5.0/gems/chef-umami-0.0.6/bin/umami:7:in `<top (required)>'
	10: from /root/.chefdk/gem/ruby/2.5.0/gems/chef-umami-0.0.6/lib/chef-umami/runner.rb:98:in `run'
	 9: from /root/.chefdk/gem/ruby/2.5.0/gems/chef-umami-0.0.6/lib/chef-umami/client.rb:43:in `compile'
	 8: from /root/.chefdk/gem/ruby/2.5.0/gems/chef-13.12.14/lib/chef/client.rb:514:in `setup_run_context'
	 7: from /root/.chefdk/gem/ruby/2.5.0/gems/chef-13.12.14/lib/chef/policy_builder/policyfile.rb:190:in `setup_run_context'
	 6: from /root/.chefdk/gem/ruby/2.5.0/gems/chef-13.12.14/lib/chef/run_context.rb:191:in `load'
	 5: from /root/.chefdk/gem/ruby/2.5.0/gems/chef-13.12.14/lib/chef/run_context/cookbook_compiler.rb:77:in `compile'
	 4: from /root/.chefdk/gem/ruby/2.5.0/gems/chef-13.12.14/lib/chef/run_context/cookbook_compiler.rb:160:in `compile_recipes'
	 3: from /root/.chefdk/gem/ruby/2.5.0/gems/chef-13.12.14/lib/chef/run_context/cookbook_compiler.rb:160:in `each'
	 2: from /root/.chefdk/gem/ruby/2.5.0/gems/chef-13.12.14/lib/chef/run_context/cookbook_compiler.rb:163:in `block in compile_recipes'
	 1: from /root/.chefdk/gem/ruby/2.5.0/gems/chef-13.12.14/lib/chef/run_context.rb:342:in `load_recipe'
/root/.chefdk/gem/ruby/2.5.0/gems/chef-13.12.14/lib/chef/cookbook_version.rb:194:in `load_recipe': could not find recipe default for cookbook santosh (Chef::Exceptions::RecipeNotFound)

**Comment by Santosh-Y-Jadhav** [comment]
Do we found any solution for it

**Comment by RyanFrantz** [comment]
I have not dug into this, but I will make time in the coming weeks to do so.

**Comment by Santosh-Y-Jadhav** [comment]
While dig down issues, Instead on using chef-zero, I have used actual chef-server 12 and modified code with latest method names used by chef_client.resource_collection. resource.state name is change to resource.state_for_resource_reporter in chef 14.0. After making this changes I am able to run you provided example to create unit and integration tests generation. I am nor sure how it was working with chef-zero? because I read couple of document it got information Currently Chef-Zero does not support the new native Policyfile endpoints. Can you confirm? https://geekblood.wordpress.com/2015/05/01/chef-server-12-and-aws-provisioning-refresh-part-1/

**Comment by RyanFrantz** [comment]
The issue here is that the dependencies for `chef-umami` are very outdated. As a result, older libraries aren't aware of the `all_files` key in newer versions of `CookbookVersion` code. That behavior is defined in [Chef RFC 67](https://github.com/chef-boneyard/chef-rfc/blob/master/rfc067-cookbook-segment-deprecation.md).

To address this, I'm working on a new version that pulls in newer `chef-dk` and kin. As a result I've discovered an unrelated bug that I'm currently working on.

Hang tight!

**Comment by RyanFrantz** [comment]
I'm closing in on a fix! I should have something ready in a day or two.

**Comment by RyanFrantz** [comment]
Fixed in #28
